src/zero_copy/input.rs

```diff
@@ -161,6 +173,32 @@ pub struct InputRef<'a, 'parse, I: Input + ?Sized, E: Error<I>, S> {
     errors: Vec<E>,
 }
 
+impl<'a, 'parse, I: Slice + ?Sized, E: Error<I>, S> InputRef<'a, 'parse, I, E, S> {
+    // used for parser combinators that want to temporarily shrink the input
+    pub(crate) fn with_subslice<T, F>(&mut self, range: Range<Marker<I>>, f: F) -> T where
+        F: Fn(&mut Self) -> T
+    {
+        let mut input = self.slice(range);
+        let mut marker = Marker {
+            pos: 0,
+            offset: input.start(),
```

Comment:
We need to be a bit careful here: `offset` is used for prioritising errors, and this effectively ends up lying to the rest of the crate about the offset of an error, which could have strange ramifications.